tensorflow/python/ops/nn_impl.py

```diff
@@ -469,9 +469,9 @@ def sufficient_statistics(x, axes, shift=None, keep_dims=False, name=None):
         counts *= x_shape[d].value
       counts = constant_op.constant(counts, dtype=x.dtype)
     else:  # shape needs to be inferred at runtime.
```

Comment:
Not sure if it would help with the original issue, but it might also be worth making L466 more permissive. e.g. Instead of:

```
if x_shape.is_fully_defined():
```

...we could instead have:

```
if all(x_shape[d].value is not None for d in axes):
```